memdb.go

```diff
@@ -95,6 +95,44 @@ func (db *MemDB) Snapshot() *MemDB {
 	return clone
 }
 
+// CreateIndexes is used for create indexes after the database already exists.
+func (db *MemDB) CreateIndexes(table string, schema ...*IndexSchema) error {
+	// lock the writer but create the transaction afterwards
+	db.writer.Lock()
+
+	// create a edge for each index to create
+	root := db.getRoot()
+	for _, indexSchema := range schema {
+		index := iradix.New()
+		path := indexPath(table, indexSchema.Name)
+		root, _, _ = root.Insert(path, index)
```

Comment:
Is there a reason why the new indexes are added to the root tree outside of the transaction below? If there are 10 indexes for example this would be 10 separate root tree transactions. I guess there is no way these insertions can "fail" so it's not really a correctness issue but leaves me wondering if there is some subtle reason I don't understand for it to be this way instead of creating the transaction above and using it?